src/modules/registration/components/registration-form/registration-form.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { Clipboard } from '@ionic-native/clipboard';
+import { Globalization } from '@ionic-native/globalization';
+import { Toast } from '@ionic-native/toast';
+import { TranslateService } from '@ngx-translate/core';
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { Print } from '@print';
+import { Registration } from '../../classes/registration';
+import { RegistrationService } from '../../providers/registration/registration';
+
+@Component({
+  selector: 'registration-form',
+  templateUrl: 'registration-form.html'
+})
+export class RegistrationFormComponent implements OnInit {
+
+  registration: Registration;
+  submitted: boolean;
+  error: any;
+
+  constructor(
+    private readonly registrationService: RegistrationService,
+    private readonly toast: Toast,
+    private readonly clipboard: Clipboard,
+    private readonly globalization: Globalization,
+    private readonly translateService: TranslateService
+  ) { }
+
+  ngOnInit() {
+    this.registration = this.registrationService.registration;
+    this.submitted = this.registrationService.userRegistered;
+  }
+
+  onSubmit(registrationForm: NgForm): void {
+    this.registrationService.sendRegistration(this.registration)
+      .subscribe(
+        (registration: Registration) => {
+          registrationForm.reset();
+          this.registration = registration;
+          this.submitted = this.registrationService.userRegistered = true;
```

Comment:
Shouldn't `userRegistered` be automatically changed in the registration service itself?